Add epoch and reconfig doc

lxfind opened the pull request:
## Description 

First draft

## Test Plan 

N/A

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes

ronny-mysten reviewed (approved):
I added some content Xun, so you might want to check it over again. Thanks!

lxfind commented:
> I added some content Xun, so you might want to check it over again. Thanks!

Thanks! I removed the ATM analogy since it might give the wrong impression on finality. Otherwise it looks great. Going to merge.
